File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 1
        right = max(piles)

        result = right

        while left <= right:
            k = (left + right) // 2
            totalTime = 0

            for p in piles:
                totalTime += math.ceil(float(p) / k)
            if totalTime <= h:
                result = k
                right = k - 1
            else:
                left = k + 1
        return result

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int area = 0;
        int max = 0;
        for (int left = 0; left < heights.length; left++) {
            for (int right = 0; right < heights.length; right++) {
                int width = right - left;
                int height = Math.min(heights[right], heights[left]);
                area =  width * height;
                max = Math.max(area, max);
            }
        }
        
        return max;
    }
}
